diff --git a/config/BuildSystem/config/packages/bison.py b/config/BuildSystem/config/packages/bison.py
index 9243a2e6b2a..4bc0cbda536 100644
--- a/config/BuildSystem/config/packages/bison.py
+++ b/config/BuildSystem/config/packages/bison.py
@@ -7,7 +7,6 @@ class Configure(config.package.GNUPackage):
     self.version           = '3.8.2'
     self.download          = ['https://ftp.gnu.org/gnu/bison/bison-'+self.version+'.tar.gz',
                               'http://mirrors.kernel.org/gnu/bison/bison-'+self.version+'.tar.gz']
-    self.lookforbydefault  = 1
     self.haveBison3plus    = 0
     self.publicInstall     = 0 # always install in PETSC_DIR/PETSC_ARCH (not --prefix) since this is not used by users
     self.executablename    = 'bison'
