diff --git a/config/BuildSystem/config/setCompilers.py b/config/BuildSystem/config/setCompilers.py
index 58c9c43d4bb..91df5fae8f2 100644
--- a/config/BuildSystem/config/setCompilers.py
+++ b/config/BuildSystem/config/setCompilers.py
@@ -1960,7 +1960,7 @@ class Configure(config.base.Configure):
     substrings = ('unknown argument', 'ignoring unsupported linker flag', 'unrecognized command line option','unrecognised command line option',
                   'unrecognized option','unrecognised option','not recognized',
                   'not recognised','unknown option','unknown warning option',
-                  'unknown flag','unknown switch','ignoring option','ignored','argument unused',
+                  'unknown flag','unknown switch','argument unused',
                   'unsupported command line options encountered',
                   'not supported','is unsupported and will be skipped','illegal option',
                   'invalid option','invalid suboption','bad ',' option','petsc error',
