diff --git a/include/petsc/private/vecimpl.h b/include/petsc/private/vecimpl.h
index dd75dbbc00b..79e40639399 100644
--- a/include/petsc/private/vecimpl.h
+++ b/include/petsc/private/vecimpl.h
@@ -110,20 +110,6 @@ struct _VecOps {
   PetscErrorCode (*setvaluescoo)(Vec, const PetscScalar[], InsertMode);
 };
 
-#if defined(offsetof) && (defined(__cplusplus) || (PETSC_C_VERSION >= 11))
-  #if (PETSC_C_VERSION >= 11) && (PETSC_C_VERSION < 23)
-    // static_assert() is a keyword since C23, before that defined as macro in assert.h
-    #include <assert.h>
-  #endif
-
-static_assert(offsetof(struct _VecOps, duplicate) == sizeof(void (*)(void)) * VECOP_DUPLICATE, "");
-static_assert(offsetof(struct _VecOps, set) == sizeof(void (*)(void)) * VECOP_SET, "");
-static_assert(offsetof(struct _VecOps, view) == sizeof(void (*)(void)) * VECOP_VIEW, "");
-static_assert(offsetof(struct _VecOps, load) == sizeof(void (*)(void)) * VECOP_LOAD, "");
-static_assert(offsetof(struct _VecOps, viewnative) == sizeof(void (*)(void)) * VECOP_VIEWNATIVE, "");
-static_assert(offsetof(struct _VecOps, loadnative) == sizeof(void (*)(void)) * VECOP_LOADNATIVE, "");
-#endif
-
 /*
     The stash is used to temporarily store inserted vec values that
   belong to another processor. During the assembly phase the stashed
